TASK: fix: ensure Prisma Client is generated on Render build

### What task?

Durante el deploy del backend en Render, la build fallaba con errores de TypeScript relacionados a Prisma, tales como:

error TS2339: Property 'product' does not exist on type 'PrismaClient'
error TS2353: Object literal may only specify known properties, and 'role' does not exist in type 'UserCreateInput'


Esto ocurría porque el cliente de Prisma no se generaba automáticamente en el entorno de Render antes de la compilación de TypeScript (tsc).

### Add screenshots

<img width="2131" height="903" alt="Image" src="https://github.com/user-attachments/assets/17da0699-d6bc-4b85-8c8f-8cacf52b8b59" />

--- a/backend/package.json
+++ b/backend/package.json
@@ -4,11 +4,11 @@
   "description": "Backend API server with Express, TypeScript, Prisma, and Swagger",
   "main": "dist/server.js",
   "scripts": {
-    "build": "tsc && tsc-alias",
+    "prisma:migrate": "prisma migrate dev",
+    "build": "npm run prisma:generate && tsc && (tsc-alias || echo 'tsc-alias not found, skipping alias fix')",
     "start": "node dist/server.js",
     "dev": "ts-node-dev --respawn --transpile-only --require tsconfig-paths/register src/server.ts",
     "prisma:generate": "prisma generate",
-    "prisma:migrate": "prisma migrate dev",
     "prisma:studio": "prisma studio",
     "prisma:seed": "ts-node prisma/seed.ts",
     "test": "jest --config jest.config.ts --runInBand",

